src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -522,20 +545,26 @@ private function fixPhpDoc(Tokens $tokens, int $index, array $uses, string $name
                 return $matches[0];
             }
 
-            // @TODO parse the complex type using TypeExpression and fix all names inside (like `int|string` or `list<int|string>`)
-            if (!Preg::match('/^[a-zA-Z0-9_\\\\]+(\|null)?$/', $matches[4])) {
-                return $matches[0];
-            }
+            /** @TODO parse the complex type using TypeExpression and fix all names inside (like `list<\Foo\Bar|'a|b|c'|string>` or `\Foo\Bar[]`) */
+            $unsupported = false;
 
-            $shortTokens = $this->determineShortType($matches[4], $uses, $namespaceName);
-            if (null === $shortTokens) {
-                return $matches[0];
-            }
+            return $matches[1].$matches[2].$matches[3].implode('|', array_map(function ($v) use ($uses, $namespaceName, &$unsupported) {
+                if ($unsupported || !Preg::match('/^'.self::REGEX_CLASS.'$/', $v)) {
+                    $unsupported = true;
+
+                    return $v;
+                }
+
+                $shortTokens = $this->determineShortType($v, $uses, $namespaceName);
+                if (null === $shortTokens) {
+                    return $v;
+                }
 
-            return $matches[1].$matches[2].$matches[3].implode('', array_map(
-                static fn (Token $token) => $token->getContent(),
-                $shortTokens
-            ));
+                return implode('', array_map(
+                    static fn (Token $token) => $token->getContent(),
+                    $shortTokens
+                ));
+            }, explode('|', $matches[4])));
```

Comment:
Will it work for `@param '/'|'|'|\Foo\StringableSeparator $separator`? Shouldn't phpDoc types be parsed with grammar?

Reply:
They should, currently the TypeExpression does not allow updates easily. I have an idea how to fix it, but I do not want make this PR even larger, so let's merge this PR and I will fix it then as a feature request - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7619#issuecomment-1890931444.